### ISSUE TITLE
Error: TypeError: Cannot call a class as a function

### ISSUE THREAD

**Issue by coolnixy** [protractor-cucumber-framework/protractor-cucumber-framework#71]
I raised the following bug in cucumber.js repo and I got  a response "It appears protractor-cucumber-framework is not compatible with the 2.0 release candidates of cucumber-js. Please create an issue there" So raising a bug here.

https://github.com/cucumber/cucumber-js/issues/755

dev dependencies in package.json

"devDependencies": {
"appium": "^1.6.3",
"babel-preset-es2015": "^6.18.0",
"babel-preset-latest": "^6.16.0",
"babel-register": "^6.18.0",
"chai": "^3.5.0",
"chai-as-promised": "^6.0.0",
"cucumber": "^2.0.0-rc.7",
"globby": "^6.1.0",
"mocha": "^3.1.2",
"mochawesome-screenshots": "^1.4.7",
"protractor": "^4.0.11",
"protractor-cucumber-framework": "^0.6.0",
"rimraf": "^2.5.4",
"wd": "^1.1.1",
"wd-bridge": "0.0.2",
"yargs": "^6.5.0"
}

My protractor cucumber config file

exports.config = {
framework: 'custom',
frameworkPath: require.resolve('protractor-cucumber-framework'),
specs: ['./specs/protractor-cucumber/features/.feature'],
cucumberOpts: {
require: './specs/protractor-cucumber/step-definitions/.js',
tags: false,
format: 'pretty',
profile: false,
'no-source': true
},
capabilities: {
browserName: 'chrome'
},
getPageTimeout: 60000 * 2,
allScriptsTimeout: 60000 * 2,

onPrepare: function() {

    require('babel-register')({
        "presets": ["latest"]
    });

    let chai = require('chai'),            
        globby = require('globby'),
        rimraf = require('rimraf');

    chai.should();
    chai.use(require('chai-as-promised'));
    global.EC = protractor.ExpectedConditions;        
    browser.driver.manage().window().maximize();
}
}

When I run the protractor protractor.cucumber.config, I get error

Error: TypeError: Cannot call a class as a function
at exports.default (D:\Apps.UI.Tests.Copy\node_modules\babel-runtime\helpers\classCallCheck.js:7:11)
at Object.Cli (D:\Apps.UI.Tests.Copy\node_modules\cucumber\lib\cli\index.js:64:34)
at D:\Apps.UI.Tests.Copy\node_modules\protractor-cucumber-framework\index.js:31:16
at Function.promise (D:\Apps.UI.Tests.Copy\node_modules\q\q.js:682:9)
at D:\Apps.UI.Tests.Copy\node_modules\protractor-cucumber-framework\index.js:24:14
at _fulfilled (D:\Apps.UI.Tests.Copy\node_modules\q\q.js:834:54)
at self.promiseDispatch.done (D:\Apps.UI.Tests.Copy\node_modules\q\q.js:863:30)
at Promise.promise.promiseDispatch (D:\Apps.UI.Tests.Copy\node_modules\q\q.js:796:13)
at D:\Apps.UI.Tests.Copy\node_modules\q\q.js:556:49
at runSingle (D:\Apps.UI.Tests.Copy\node_modules\q\q.js:137:13)
at flush (D:\Apps.UI.Tests.Copy\node_modules\q\q.js:125:13)
at _combinedTickCallback (internal/process/next_tick.js:67:7)
at process._tickCallback (internal/process/next_tick.js:98:9)
[16:51:53] E/launcher - Process exited with error code 100

**Comment by mattbrunetti** [comment]
@coolnixy This is a duplicate of #64 (which could probably be renamed as "Support for cucumber v2" some similar). 

There's been some work at resolving this in PR #66 by @darrinholst.

I believe the main thing we're waiting for is (or as far as I know, the only thing we need is) a way to test against and support both versions of Cucumber in the same package. See discussion on issue #64. @darrinholst Is there anything else we want/need to add before we can merge your PR?

I tried to do a little work on that (testing against both versions of Cucumber), but ran into some difficulty just setting up the package in dev mode. Either something to do with developing on Windows, or setting up Selenium, or a combination. :/

**Comment by darrinholst** [comment]
That pr is not waiting on anything. It's backwards compatible between cucumber 1 and 2. It would be nice to have tests for both versions, but it's not mandatory.

**Comment by igniteram** [comment]
@darrinholst Can you pls merge the PR? I have tested it against both the cucumber versions 1 & 2 and with protractor 5(latest) it is working fine. I am working on the tests, soon I would be sending a PR which would have tests ,examples etc. with latest & older versions of protractor,cucumber. If the users face any issue they could raise an issue and we could address them. We should not be holding off https://github.com/mattfritz/protractor-cucumber-framework/pull/66 for so long.

**Comment by darrinholst** [comment]
@igniteram I wish I could push the button, but afaik @mattfritz is the only one with commit rights here and he's gone dark 😉

**Comment by coolnixy** [comment]
Thanks guys for your response. Now it's just a wait before @mattfritz commits the changes.

**Comment by darrinholst** [comment]
You can use that pr branch directly from github in the meantime. That's what I've been doing.

        "protractor-cucumber-framework": "darrinholst/protractor-cucumber-framework#cucumber-2",

**Comment by darrinholst** [comment]
I published this under https://www.npmjs.com/package/%40darrinholst%2Fprotractor-cucumber-framework

**Comment by mattbrunetti** [comment]
@darrinholst Thank you for your work!

Might want to update `repository.url` in the package.json, so that the page on npmjs.com points to the right repository.

**Comment by igniteram** [comment]
@darrinholst Thanks darrin, appreciate your efforts!

**Comment by darrinholst** [comment]
cucumber 2 support is in v1.0.0 now
